axelor-supplychain/src/main/java/com/axelor/apps/supplychain/service/PurchaseOrderInvoiceServiceImpl.java

```diff
@@ -729,4 +729,17 @@ public void displayErrorMessageIfPurchaseOrderIsInvoiceable(
           purchaseOrder.getPurchaseOrderSeq());
     }
   }
+
+  @Override
+  public boolean containsRelatedVentilatedInvoice(PurchaseOrder purchaseOrder) {
+    List<Invoice> relatedVentilatedInvoiceList =
+        invoiceRepo
+            .all()
+            .filter(
+                "self.purchaseOrder.id = ?1 AND self.statusSelect = ?2",
+                purchaseOrder.getId(),
+                InvoiceRepository.STATUS_VENTILATED)
+            .fetch();
+    return !relatedVentilatedInvoiceList.isEmpty();
```

Comment:
Please use `count` instead of `fetch` and compare it to 0. Also, please use `bind` with variables names instead of `?` for better readability